alacritty/src/ipc.rs

```diff
@@ -55,9 +86,13 @@ pub fn spawn_ipc_socket(options: &Options, event_proxy: EventLoopProxy<Event>) -
 }
 
 /// Send a message to the active Alacritty socket.
-pub fn send_message(socket: Option<PathBuf>, message: &[u8]) -> IoResult<()> {
-    let socket = find_socket(socket)?;
-    socket.send(message)?;
+pub fn send_message(socket: Option<PathBuf>, message: IPCMessage) -> IoResult<()> {
+    let mut socket = find_socket(socket)?;
+
+    let mut message = serde_json::to_string(&message)?;
+    message.push(JSON_SEPARATOR as char);
+    socket.write_all(message[..].as_bytes())?;
```

Comment:
Is it actually necessary to push the `JSON_SEPARATOR` here? Seems somewhat strange from an API perspective for Rust, doesn't `flush` work?